maven-core/src/main/java/org/apache/maven/internal/transformation/ArtifactTransformer.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.internal.transformation;
+
+import java.io.IOException;
+import org.apache.maven.project.MavenProject;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.deployment.DeployRequest;
+import org.eclipse.aether.installation.InstallRequest;
+
+/**
+ * Artifact transformer component.
+ * <p>
+ * Note: this interface is Maven internal component, subject to change or removal without notice.
+ *
+ * @since TBD
+ */
+public interface ArtifactTransformer {
```

Comment:
If we don't foresee any additional usage of this interface, I'd rather simply remove it and simply inject the `ConsumerPomArtifactTransformer`.

Reply:
Done

Reply:
Is the pom an artifact or is it the metadata for the artifact and subartifacts? Just wondering if the class has the right name. (hence why i called it FileTransformer to cover both)

Reply:
POM is attached as Artifact (but this interface was removed since)